lib/chef/provider/package/chocolatey.rb

```diff
@@ -141,6 +154,39 @@ def query_command
           "search"
         end
 
+        # invalidate cache for testing purposes
+        def invalidate_cache
+          @@choco_config = nil
+        end
+
+        # This checks that the repo list has not changed between now and when we last checked
+        # the cache
+        def cache_is_valid?
+          return false if @@choco_config.nil? || (actual_config != @@choco_config)
+
+          true
+        end
+
+        # walk the collection to find peer resources to ensure that
+        # we get as much of the cache this run as possible.  Unified mode
+        # sub-resources will, of course, be incremental with this, since we can't
+        # grab them in advance, but even in that case we're still way, way
+        # more efficient with our queries...
+        #
+        # @return [Array] List of chocolatey packages referenced in the run list
+        def collect_package_requests(ignore_list: [])
+          return ['*'] if new_resource.bulk_query || ENV['BULK_QUERY_CHOCOLATEY'] == 'true'
```

Comment:
The env var is an anti-pattern. You could make a Chef::Config option if you want something more global.